staging/src/k8s.io/kubectl/pkg/cmd/config/set_context.go

```diff
@@ -79,6 +79,12 @@ func NewCmdConfigSetContext(out io.Writer, configAccess clientcmd.ConfigAccess)
 	cmd.Flags().Var(&options.cluster, clientcmd.FlagClusterName, clientcmd.FlagClusterName+" for the context entry in kubeconfig")
 	cmd.Flags().Var(&options.authInfo, clientcmd.FlagAuthInfoName, clientcmd.FlagAuthInfoName+" for the context entry in kubeconfig")
 	cmd.Flags().Var(&options.namespace, clientcmd.FlagNamespace, clientcmd.FlagNamespace+" for the context entry in kubeconfig")
+	cmdutil.CheckErr(cmd.RegisterFlagCompletionFunc(
```

Comment:
We're actually setting:
```
ValidArgsFunction: completion.ContextCompletionFunc
```
We can't override it right?

Reply:
> I don't think that's possible to replace these two, because the added behavior only runs completion for `namespace` flag. But i think it's more cleaner IMO to move this section inside `ContextCompletionFunc`, this looks good too.

@ah8ad3  Doesn't that mean that each time there's a request for an autocompletion, a flag registration would be executed?

Reply:
@ah8ad3 You mean like this, right?

```
func ContextCompletionFunc(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
	cmdutil.CheckErr(cmd.RegisterFlagCompletionFunc(
		"namespace",
		func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
			return CompGetResource(factory, "namespace", toComplete), cobra.ShellCompDirectiveNoFileComp
		},
	))

	if len(args) == 0 {
		return ListContextsInConfig(toComplete), cobra.ShellCompDirectiveNoFileComp
	}
	return nil, cobra.ShellCompDirectiveNoFileComp
}
```

Reply:
> Doesn't that mean that each time there's a request for an autocompletion, a flag registration would be executed?

Sorry for late response i was so busy, think you are right, my bad. My intention was to define all completion related stuff in one place, maybe there is a way to achieve that or maybe not.

Reply:
@ardaguclu any comment on this?

Reply:
I think, current version of this PR looks good